packages/tooling/fast-tooling/src/web-components/html-render-layer-inline-edit/html-render-layer-inline-edit.ts

```diff
@@ -0,0 +1,226 @@
+import { customElement, observable } from "@microsoft/fast-element";
+import { MessageSystemDataTypeAction, MessageSystemType } from "../../message-system";
+import {
+    ActivityType,
+    HTMLRenderLayer,
+    OverylayPosition,
+} from "../html-render-layer/html-render-layer";
+import { HTMLRenderOriginatorId } from "../html-render/html-render";
+import { HTMLRenderLayerInlineEditStyles } from "./html-render-layer-inline-edit.style";
+import { HTMLRenderLayerInlineEditTemplate } from "./html-render-layer-inline-edit.template";
+
+@customElement({
+    name: "fast-tooling-html-render-layer-inline-edit",
+    template: HTMLRenderLayerInlineEditTemplate,
+    styles: HTMLRenderLayerInlineEditStyles,
+})
+export class HTMLRenderLayerInlineEdit extends HTMLRenderLayer {
+    public layerActivityId: string = "InlineEditLayer";
+
+    @observable
+    public textAreaActive: boolean = false;
+
+    @observable
+    public textPosition: OverylayPosition = new OverylayPosition(0, 0, 0, 0);
+
+    @observable
+    public textValue: string = "";
+
+    public textAreaRef: HTMLTextAreaElement;
+
+    private currentDataId: string = null;
+    private originalTextValue: string = null;
+    private currentStyleTarget: HTMLElement;
+    private currentTextNode: Node;
+    private timeoutRef: number = null;
+
+    connectedCallback() {
+        super.connectedCallback();
+
+        window.addEventListener("scroll", this.handleWindowChange);
+    }
+
+    disconnectedCallback() {
+        super.disconnectedCallback();
+
+        window.removeEventListener("scroll", this.handleWindowChange);
+    }
+
+    private handleWindowChange = () => {
+        if (this.textAreaActive) {
+            if (this.timeoutRef !== null) {
+                window.clearTimeout(this.timeoutRef);
+            }
+            this.timeoutRef = window.setTimeout(() => {
+                this.applySizeAndPositionToTextbox();
+            }, 40);
+        }
+    };
+
+    public handleKeyDown(e: KeyboardEvent) {
+        if (e.key.length === 1) {
+            this.currentTextNode.textContent += e.key === " " ? " " : e.key;
+            this.applySizeAndPositionToTextbox();
+        }
+        return true;
+    }
+
+    public handleTextInput(e: KeyboardEvent) {
+        if (e.key === "Enter") {
+            this.commitEdit();
+            e.preventDefault();
+            return false;
+        } else if (e.key === "Escape") {
+            this.cancelEdit();
+            e.preventDefault();
+            return false;
+        }
+        const inputVal = (e.composedPath()[0] as HTMLInputElement).value;
+        this.currentTextNode.textContent = inputVal;
+        this.applySizeAndPositionToTextbox();
+        if (this.activityCallback) {
+            this.activityCallback(this.layerActivityId, ActivityType.update);
+        }
+    }
+
+    public handleBlur(e: InputEvent) {
+        this.cancelEdit();
+    }
+
+    private GetPositionFromElement(target: Node): OverylayPosition {
```

Comment:
Just noticed this was capitalized for some reason.

Reply:
Fixed